### PR TITLE
Tag and omit references to the refs file

### PR DESCRIPTION
**Pull request by puerco**
We now tag and emit references into the refs file for all arch images to help external
tools know what we've built (for example to sign  them).

Closes: https://github.com/chainguard-dev/apko/issues/288

/cc @mattmoor 

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

### CODE REVIEW

File: internal/cli/publish.go
Comment (mattmoor) [inline internal/cli/publish.go:248]: I'm curious why we need this part of the change, or whether it regresses something 🤔

**Review by mattmoor — APPROVED** [review]

**Comment by mattmoor** [comment]
Once this lands, this downstream change will get pretty interesting 😉 

https://github.com/distroless/static/pull/11

File: internal/cli/publish.go
Comment (puerco) [reply internal/cli/publish.go:248]: Do you think it's better if I dont pass it to the tagging function and build the reference by hand? I did passed it because it was returning an empty string .

Comment (mattmoor) [reply internal/cli/publish.go:248]: I don't have enough context here, so I'd defer to @kaniini

Comment (puerco) [reply internal/cli/publish.go:248]: OK, I thought you had a specific problem in mind. I think we're fine, tests are ok and I've been using this branch today ok.

**Pull request merged**
